### PR TITLE
Increase GPT4o context size budget

### PR DESCRIPTION
**Pull request by philipp-spiess**
Is there any server-side changes that we need to make here? If nobody knows from the top of their head, I will check it myself before merging.

See: https://nian.llmonpy.ai/ on data for gpt4o recall

## Test plan

- CI will test

### CODE REVIEW

File: lib/shared/src/models/dotcom.ts [hunk 1]
```diff
@@ -20,11 +20,11 @@ const DEFAULT_DOT_COM_MODELS: ModelProvider[] = [
         initialDefault: true,
         codyProOnly: false,
         usage: [ModelUsage.Chat, ModelUsage.Edit],
-        // Has a higher context window with a seperated limit for user-context.
+        // Has a higher context window with a separated limit for user-context.
```
Comment (dominiccooney) [inline lib/shared/src/models/dotcom.ts:23]:
```suggestion
        // Has a higher context window with a separate limit for user-context.
```

File: lib/shared/src/models/dotcom.ts [hunk 2]
```diff
@@ -36,11 +36,11 @@ const DEFAULT_DOT_COM_MODELS: ModelProvider[] = [
         default: false,
         codyProOnly: true,
         usage: [ModelUsage.Chat, ModelUsage.Edit],
-        // Has a higher context window with a seperated limit for user-context.
+        // Has a higher context window with a separated limit for user-context.
```
Comment (dominiccooney) [inline lib/shared/src/models/dotcom.ts:39]:
```suggestion
        // Has a higher context window with a separate limit for user-context.
```

File: lib/shared/src/models/dotcom.ts [hunk 3]
```diff
@@ -63,7 +63,12 @@ const DEFAULT_DOT_COM_MODELS: ModelProvider[] = [
         default: false,
         codyProOnly: true,
         usage: [ModelUsage.Chat, ModelUsage.Edit],
-        contextWindow: { input: CHAT_INPUT_TOKEN_BUDGET, output: CHAT_OUTPUT_TOKEN_BUDGET },
+        // Has a higher context window with a separated limit for user-context.
```
Comment (dominiccooney) [inline lib/shared/src/models/dotcom.ts:66]:
```suggestion
        // Has a higher context window with a separate limit for user-context.
```

File: lib/shared/src/token/constants.ts [hunk 4]
```diff
@@ -20,15 +20,15 @@ export const CHAT_OUTPUT_TOKEN_BUDGET = 4000
 export const ENHANCED_CONTEXT_ALLOCATION = 0.6
 
 /**
- * NOTE: Reserved for Claude-3 Sonnet and Opus only.
+ * NOTE: Reserved for models with high context windows and good recall.
```
Comment (dominiccooney) [inline lib/shared/src/token/constants.ts:23]:
```suggestion
 * NOTE: Reserved for models with large context windows and good recall.
```

File: lib/shared/src/token/constants.ts [hunk 5]
```diff
@@ -20,15 +20,15 @@ export const CHAT_OUTPUT_TOKEN_BUDGET = 4000
...
  *
  * The total context window reserved for user added context (@-mention, right-click, etc.)
  */
-export const CLAUDE3_USER_CONTEXT_TOKEN_BUDGET = 30000
+export const EXTENDED_USER_CONTEXT_TOKEN_BUDGET = 30000
 
 /**
- * NOTE: Reserved for Claude-3 Sonnet and Opus only.
+ * NOTE: Reserved for models with high context windows and good recall.
```
Comment (dominiccooney) [inline lib/shared/src/token/constants.ts:30]:
```suggestion
 * NOTE: Reserved for models with large context windows and good recall.
```

**Review by dominiccooney — APPROVED** [review]
Looks great. Some grammar nits inline.

File: lib/shared/src/models/dotcom.ts (hunk 1 above)
Comment (philipp-spiess) [reply lib/shared/src/models/dotcom.ts:23]: wow this is what you get when you attempt to fix 1 typo 😆

**Comment by philipp-spiess** [comment]
No other backend concerns. Showtime! https://sourcegraph.slack.com/archives/C05497E9MDW/p1715878673939979?thread_ts=1715851421.420119&cid=C05497E9MDW

**Pull request merged**